disnake/guild.py

```diff
@@ -3812,11 +3836,25 @@ async def create_role(
         else:
             fields["permissions"] = "0"
 
-        actual_colour = colour or color or Colour.default()
-        if isinstance(actual_colour, int):
-            fields["color"] = actual_colour
-        else:
-            fields["color"] = actual_colour.value
+        actual_primary_color = colour or color or primary_colour or primary_color
+        actual_secondary_color = secondary_colour or secondary_color
+        actual_tertiary_color = tertiary_colour or tertiary_color
```

Comment:
Similar comment about `#000000` being falsy here. Handling this makes things much more lengthy, but it would be more correct.